src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -349,86 +349,182 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
      */
-    fluid.transforms.arrayToSetMembership.invertWithType = function (transformSpec, transformer, newType) {
+    fluid.transforms.arrayToSetMembership.invertWithType = function (transformSpec, newType) {
         transformSpec.type = newType;
         var newOptions = {};
         fluid.each(transformSpec.options, function (path, oldKey) {
             newOptions[path] = oldKey;
         });
-        transformSpec.options = newOptions;
+        if (!$.isEmptyObject(newOptions)) {
+            transformSpec.options = newOptions;
+        }
         return transformSpec;
     };
 
-    fluid.transforms.arrayToSetMembership.invert = function (transformSpec, transformer) {
-        return fluid.transforms.arrayToSetMembership.invertWithType(transformSpec, transformer,
-            "fluid.transforms.setMembershipToArray");
-    };
-
     fluid.defaults("fluid.transforms.setMembershipToArray", {
         gradeNames: ["fluid.standardTransformFunction", "fluid.lens"],
         invertConfiguration: "fluid.transforms.setMembershipToArray.invert"
     });
 
-    fluid.transforms.setMembershipToArray = function (input, transformSpec, transformer) {
-        var options = transformSpec.options;
-
-        if (!options) {
-            fluid.fail("setMembershipToArray requires an options block specified");
-        }
-
-        if (transformSpec.presentValue === undefined) {
-            transformSpec.presentValue = true;
-        }
-
-        if (transformSpec.missingValue === undefined) {
-            transformSpec.missingValue = false;
+    /**
+     *
+     * Transform the input object into an array based upon the options provided to the specification. Accepts
+     * an object as its first argument and an optional second argument (an object) which contains the
+     * specifications for the resulting output.
+     *
+     * An error will be thrown if the input to be transformed is not an object.
+     *
+     * Partly invertible (into arrayToSetMembership)
+     *
+     * For example, with the following transformSpec:
+     *
+     * {
+     *   "options": {
+     *      "hasMouse": "mouse",
+     *      "hasKeyboard": "keyboard",
+     *      "hasTrackpad": "trackpad",
+     *      "hasHeadtracker": "headtracker"
+     *   },
+     *   "presentValue": "supported",
+     *   "missingValue": "not supported"
+     * }
+     *
+     * For an input object like this:
+     *
+     * {
+     *   "hasMouse": "supported",
+     *   "hasKeyboard": "supported",
+     *   "hasTrackpad": "not supported",
+     *   "hasHeadtracker": "not supported"
+     * }
+     *
+     * The output array will be:
+     *
+     * ["keyboard", "mouse"]
+     *
+     * https://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#fluidtransformssetmembershiptoarray
+     *
+     * @param {Object} input - The object to be transformed into an array.
+     * @param {Object} [transformSpec] - The options provided to the transformation rule.
```

Comment:
Whilst it isn't necessary, just for clarity for those unused to the JSDoc convention we tend to write (optional) after the hyphen for optional arguments, in addition to placing their name in square brackets